utils/test_http_server/http_server_test_funcs.py

```diff
@@ -0,0 +1,167 @@
+import requests
+import logging
+import json
+from datetime import datetime
+import time
+
+# FUTURE TESTS IDEAS:
+# admin valid: '{"connect_ip":"10.24.230.10","connect_key":"01234567890123456789","connect_name":"P3D"}'
+# admin not valid ip: '{"connect_ip":"1000.24.230.10","connect_key":"11234567890123456789","connect_name":"PRUSA3D"}'
+# admin longer str: '{"connect_ip":"10.24.230.10","connect_key":"11114567890123456789123456","connect_name":"PRUSA3D01234567890123456789"}'
+
+# global variables
+IP_ADDR = ""
+test_cnt = 0
+tests_off = 0
+test_curr = 0
+json_tests = []
+
+# gcode structures
+G_AUTOHOME = {"command":"G28"}
+G_MOVES = {"command":"G1 X50 Y100", "command":"G1 X100 Y50", "command":"G1 X60 Y90", "command":"G1 X100 Y50", "command":"G1 X70 Y80"}
+
+def init(ip_addr):
+    global json_tests, test_cnt, IP_ADDR
+
+    # set ip address
+    IP_ADDR = 'http://' + ip_addr
+
+    # load JSON tests
+    json_file = open("server_tests/tests.json", "r")
+    json_obj = json.load(json_file)
+    json_file.close()
+
+    json_tests = json_obj['tests']
+    disabled_tests = []
+
+    # disable switched off tests
+    for test in json_tests:
+        if "switch" in test:
+            if "off" in test["switch"]:
+                disabled_tests.append(test)
+    
+    for test in disabled_tests:
+        json_tests.remove(test)
+
+    if len(json_tests) == 0:
+        print("All tests switched off")
+    
+    test_cnt = len(json_tests)
+
+    # logging init
+    logging.basicConfig(filename='server_tests/errors.log', filemode='w', level=logging.ERROR)
+
+# infinite testing loop
+# if all tests in tests.json are switched off, it tests only telemetry
+def test_loop():
+    global test_cnt
+
+    while 1:
+        test_get_telemetry()
+        if test_cnt != 0:
+            test()
+
+        time.sleep(1)
+
+def test():
+    global json_tests, test_curr
+
+    # send request according to test request data
+    result_dic = send_request()
+    if len(result_dic) != 0:
+        # test response 
+        test_response(result_dic)
+
+    # set up next test
+    if (test_curr + 1) >= test_cnt :
+        test_curr = 0
+        off_test_cnt = 0
+    else:
+        test_curr += 1
+
+def send_request():
+    global test_curr, test_cnt, json_tests, IP_ADDR
+    
```

Comment:
there is no need to parse content of the test, in the tests are prepared the header and body. here you send it

Reply:
I dont understand how you mean this. Header is not prepared in tests.json, it has to be parsed from json to use in requests. Body is in tests.json prepared as a json structure and it's only passed in requests...